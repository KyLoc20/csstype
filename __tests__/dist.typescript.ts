import * as path from 'path';

const COMPILER_OPTIONS = {
  noEmit: true,
  strict: true,
};

describe('Typescript 3.7', () => {
  it('detects errors', async () => {
    const ts = await import('typescript');
    const program = ts.createProgram([path.resolve(__dirname, '__fixtures__/typecheck.ts')], COMPILER_OPTIONS);
    const diagnostics = ts.getPreEmitDiagnostics(program);
    const errors = diagnostics.map(diagnostic => {
      const { line, character } = diagnostic.file!.getLineAndCharacterOfPosition(diagnostic.start!);
      return `${line}:${character} - ${ts.flattenDiagnosticMessageText(diagnostic.messageText, '\n')}`;
    });

    expect(Number(ts.versionMajorMinor)).toBe(3.7);
    expect(errors.length).toBe(14);
    for (const error of errors) {
      expect(error).toMatchSnapshot();
    }
  });
});

describe('Typescript 3.6', () => {
  it('detects errors', async () => {
    const ts = await import('typescript3.6');
    const program = ts.createProgram([path.resolve(__dirname, '__fixtures__/typecheck.ts')], COMPILER_OPTIONS);
    const diagnostics = ts.getPreEmitDiagnostics(program);
    const errors = diagnostics.map(diagnostic => {
      const { line, character } = diagnostic.file!.getLineAndCharacterOfPosition(diagnostic.start!);
      return `${line}:${character} - ${ts.flattenDiagnosticMessageText(diagnostic.messageText, '\n')}`;
    });

    expect(Number(ts.versionMajorMinor)).toBe(3.6);
    expect(errors.length).toBe(14);
    for (const error of errors) {
      expect(error).toMatchSnapshot();
    }
  });
});

describe('Typescript 3.5', () => {
  it('detects errors', async () => {
    const ts = await import('typescript3.5');
    const program = ts.createProgram([path.resolve(__dirname, '__fixtures__/typecheck.ts')], COMPILER_OPTIONS);
    const diagnostics = ts.getPreEmitDiagnostics(program);
    const errors = diagnostics.map(diagnostic => {
      const { line, character } = diagnostic.file!.getLineAndCharacterOfPosition(diagnostic.start!);
      return `${line}:${character} - ${ts.flattenDiagnosticMessageText(diagnostic.messageText, '\n')}`;
    });

    expect(Number(ts.versionMajorMinor)).toBe(3.5);
    expect(errors.length).toBe(14);
    for (const error of errors) {
      expect(error).toMatchSnapshot();
    }
  });
});
