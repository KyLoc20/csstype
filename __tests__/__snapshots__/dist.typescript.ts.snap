// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it detects errors 1`] = `
"42:2 - Type '{ unknownProperty: number; }' is not assignable to type 'Properties<0 | (string & {})>'.
  Object literal may only specify known properties, and 'unknownProperty' does not exist in type 'Properties<0 | (string & {})>'."
`;

exports[`it detects errors 2`] = `
"46:2 - Type 'number[]' is not assignable to type '0 | \\"auto\\" | \\"inherit\\" | (string & {}) | \\"-moz-initial\\" | \\"initial\\" | \\"revert\\" | \\"unset\\" | undefined'.
  Type 'number[]' is not assignable to type '\\"unset\\"'."
`;

exports[`it detects errors 3`] = `
"47:2 - Type 'number[]' is not assignable to type '\\"inherit\\" | \\"-moz-initial\\" | \\"initial\\" | \\"revert\\" | \\"unset\\" | (number & {}) | undefined'.
  Type 'number[]' is not assignable to type '\\"unset\\"'."
`;

exports[`it detects errors 4`] = `
"52:6 - Type '{}' is not assignable to type 'string & {}'.
  Type '{}' is not assignable to type 'string'."
`;
